packages/flutter_tools/lib/src/windows/migrations/version_migration.dart

```diff
@@ -0,0 +1,97 @@
+
+const String _cmakeFileBefore =
+'# Apply the standard set of build settings. This can be removed for applications\r\n'
+'# that need different build settings.\r\n'
+'apply_standard_settings(\${BINARY_NAME})\r\n'
+'\r\n'
+'# Disable Windows macros that collide with C++ standard library functions.\r\n'
+'target_compile_definitions(\${BINARY_NAME} PRIVATE "NOMINMAX")\r\n';
+const String _cmakeFileAfter =
+'# Apply the standard set of build settings. This can be removed for applications\r\n'
+'# that need different build settings.\r\n'
+'apply_standard_settings(\${BINARY_NAME})\r\n'
+'\r\n'
+'# Add preprocessor definitions for the build version.\r\n'
+'target_compile_definitions(\${BINARY_NAME} PRIVATE "FLUTTER_VERSION=\\"\${FLUTTER_VERSION}\\"")\r\n'
+'target_compile_definitions(\${BINARY_NAME} PRIVATE "FLUTTER_VERSION_MAJOR=\${FLUTTER_VERSION_MAJOR}")\r\n'
+'target_compile_definitions(\${BINARY_NAME} PRIVATE "FLUTTER_VERSION_MINOR=\${FLUTTER_VERSION_MINOR}")\r\n'
+'target_compile_definitions(\${BINARY_NAME} PRIVATE "FLUTTER_VERSION_PATCH=\${FLUTTER_VERSION_PATCH}")\r\n'
+'target_compile_definitions(\${BINARY_NAME} PRIVATE "FLUTTER_VERSION_BUILD=\${FLUTTER_VERSION_BUILD}")\r\n'
+'\r\n'
+'# Disable Windows macros that collide with C++ standard library functions.\r\n'
+'target_compile_definitions(\${BINARY_NAME} PRIVATE "NOMINMAX")\r\n';
+
+const String _resourceFileBefore =
+'#ifdef FLUTTER_BUILD_NUMBER\r\n'
+'#define VERSION_AS_NUMBER FLUTTER_BUILD_NUMBER\r\n'
+'#else\r\n'
+'#define VERSION_AS_NUMBER 1,0,0\r\n'
+'#endif\r\n'
+'\r\n'
+'#ifdef FLUTTER_BUILD_NAME\r\n'
+'#define VERSION_AS_STRING #FLUTTER_BUILD_NAME\r\n'
+'#else\r\n'
+'#define VERSION_AS_STRING "1.0.0"\r\n'
+'#endif\r\n';
+const String _resourceFileAfter =
+'#if defined(FLUTTER_VERSION_MAJOR) && defined(FLUTTER_VERSION_MINOR) && defined(FLUTTER_VERSION_PATCH) && defined(FLUTTER_VERSION_BUILD)\r\n'
+'#define VERSION_AS_NUMBER FLUTTER_VERSION_MAJOR,FLUTTER_VERSION_MINOR,FLUTTER_VERSION_PATCH,FLUTTER_VERSION_BUILD\r\n'
+'#else\r\n'
+'#define VERSION_AS_NUMBER 1,0,0,0\r\n'
+'#endif\r\n'
+'\r\n'
+'#if defined(FLUTTER_VERSION)\r\n'
+'#define VERSION_AS_STRING FLUTTER_VERSION\r\n'
+'#else\r\n'
+'#define VERSION_AS_STRING "1.0.0"\r\n'
+'#endif\r\n';
+
+// Flutter should set the Windows app's version information.
+// See https://github.com/flutter/flutter/issues/73652.
+class VersionMigration extends ProjectMigrator {
+  VersionMigration(WindowsProject project, super.logger)
+    : _cmakeFile = project.runnerCmakeFile, _resourceFile = project.runnerResourceFile;
+
+  final File _cmakeFile;
+  final File _resourceFile;
+
+  @override
+  void migrate() {
+    if (!_cmakeFile.existsSync()) {
+      logger.printTrace('windows/runner/CMakeLists.txt file not found, skipping version migration');
+      return;
+    }
+
+    if (!_resourceFile.existsSync()) {
+      logger.printTrace('windows/runner/Runner.rc file not found, skipping version migration');
+      return;
```

Comment:
to clarify, we only want to do any migrations if BOTH of these files exist?

Reply:
Yes, but we could migrate just a single file. Each file's changes are backwards compatible and would continue working even if the other file's migration wasn't applied.

Reply:
Is there a docs page we can point people to when this fails, so they can do it by hand if desired? If so, in addition, log something like "For more information see: URL".

Reply:
I added some more information to the trace to make it more actionable. This failure scenario seems unlikely (it wouldn't be unreasonable for `flutter build windows` to break if the runner's CMakeLists.txt file was missing), so I don't think it warrants a dedicated docs page.